src/main/java/seedu/address/model/food/Carbohydrate.java

```diff
@@ -10,7 +10,7 @@
      * @param amount A valid amount.
      */
     public Carbohydrate(int amount) {
-        super("Carbohydrate", amount, CARBOHYDRATE_MULTIPLIER);
+        super(Carbohydrate.class.getName(), amount, CARBOHYDRATE_MULTIPLIER);
```

Comment:
IMO we should keep this as the string "Carbohydrate" instead. Makes the code more explicit.

Same for the other macronutrients

Reply:
I think we should keep it that way to prevent potential typos. Instead of that, we can add a comment at the end of the line to explicitly state what Carbohydrate.class.getName() is

Reply:
Actually, we cannot call `this.class` before the object is initiated. Therefore, this.class.getName() cannot be put in either `Macronutrient`'s constructor or a static method.
I think we can use this.class.getName as a replacement for the macronutrientType attribute